packages/test-sync/src/commonTest/kotlin/io/realm/kotlin/test/mongodb/common/AppTests.kt

```diff
@@ -164,50 +163,51 @@ class AppTests {
     @Test
     fun allUsers() = runBlocking {
         assertEquals(0, app.allUsers().size)
+
         val user1 = app.login(Credentials.anonymous())
         var allUsers = app.allUsers()
         assertEquals(1, allUsers.size)
-        assertTrue(allUsers.containsKey(user1.identity))
-        assertEquals(user1, allUsers[user1.identity])
+        assertTrue(allUsers.contains(user1))
 
         // Only 1 anonymous user exists, so logging in again just returns the old one
         val user2 = app.login(Credentials.anonymous())
         allUsers = app.allUsers()
         assertEquals(1, allUsers.size)
-        assertTrue(allUsers.containsKey(user2.identity))
+        assertTrue(allUsers.contains(user2))
 
         val user3: User = app.asTestApp.createUserAndLogIn(TestHelper.randomEmail(), "123456")
         allUsers = app.allUsers()
         assertEquals(2, allUsers.size)
-        assertTrue(allUsers.containsKey(user3.identity))
+        assertTrue(allUsers.contains(user3))
 
         // Logging out users that registered with email/password will just put them in LOGGED_OUT state
         user3.logOut()
         allUsers = app.allUsers()
         assertEquals(2, allUsers.size)
-        assertTrue(allUsers.containsKey(user3.identity))
-        assertEquals(User.State.LOGGED_OUT, allUsers[user3.identity]!!.state)
```

Comment:
I think we should keep the LOGGED_OUT state check (maybe using `User.identities`?)